### PR TITLE
When getting an app's IAM role only make one HTTP request

### PR DESCRIPTION
**Pull request by xoen**
This is possible because we [now normalise the apps' `repo_url`]( https://github.com/ministryofjustice/analytics-platform-control-panel/pull/44) to the
version without the trailing `.git`.

Part of ticket: https://trello.com/c/yfUlKPdb/428-cp-api-jenkins-normalise-repourl-and-avoid-possible-second-request

### CODE REVIEW

File: vars/cpapi.groovy [hunk 1]
```diff
@@ -31,16 +29,12 @@ def _queryString(params = [:]) {
...
+def _getRepoURL() {
   def repoURL = gitRepo.getURL()
-  def withGit = repoURL
-  def withoutGit = repoURL
 
   if (repoURL.endsWith(".git")) {
-      withoutGit = repoURL.split("\\.git\$")[0]
-  } else {
-      withGit = repoURL + ".git"
+      return repoURL.split("\\.git\$")[0]
```
Comment (andyhd) [inline vars/cpapi.groovy]: I think Groovy supports regex literals, so you could write this as
```groovy
return repoURL.split(/\.git$/)[0]
```

Comment (andyhd) [reply vars/cpapi.groovy]: Anyway, regexs are not needed here - you can (should be able to?) just substring:
```groovy
def gitSuffix = '.git'
if (repoURL.endsWith(gitSuffix)) {
    return repoURL.substring(0, repoURL.length() - gitSuffix.length())
}
```

File: vars/cpapi.groovy [hunk 2]
```diff
@@ -31,16 +29,12 @@ def _queryString(params = [:]) {
 }
 
 
-def _getRepoURLs() {
+def _getRepoURL() {
```
Comment (andyhd) [inline vars/cpapi.groovy]: I might rename this `_normalizeRepoURL` too.

File: vars/cpapi.groovy (hunk 1 above)
Comment (xoen) [reply vars/cpapi.groovy]: OK, I'll replace with `return repoURL[0..-5]` (I'd avoid the `.length()` as chances of this logic changing are thin and even if it does change change should be trivial)

File: vars/cpapi.groovy (hunk 2 above)
Comment (xoen) [reply vars/cpapi.groovy]: Renamed to `_normalisedRepoURL()` 👍

**Comment by xoen** [comment]
Thanks Andy, I'll do the usual testing in `dev` then merge 👍

**Comment by xoen** [comment]
Cool it works as expected https://(DEV_JENKINS)/job/moj-analytical-services-github/job/secure_geezer/job/ag--dont-ask-iam-role/9/console

**Pull request merged**